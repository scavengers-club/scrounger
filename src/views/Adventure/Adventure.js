import React, { useState, useEffect } from 'react';
import { useParams } from 'react-router';
import AdventureDetail from '../../components/AdventureDetail/AdventureDetail';
import { getAdventureById } from '../../services/adventures';

export default function Adventure() {
  const [adventure, setAdventure] = useState({});
  const { id } = useParams();
  useEffect(() => {
    const fetchData = async () => {
      const resp = await getAdventureById(id);
      setAdventure(resp);
    };
    fetchData();
  }, [id]);
  return <AdventureDetail {...adventure} />;
}
