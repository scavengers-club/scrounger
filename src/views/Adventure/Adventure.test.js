import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Adventure from './Adventure';

test('<Adventure/> should render the list of adventures', async () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/adventure/1']}>
      <Route path="/adventure/:id">
        <Adventure
          currentUser={{ id: '575e91dd-2c9c-4297-aa60-6504a62236dc', username: 'violet' }}
        />
      </Route>
    </MemoryRouter>
  );

  await screen.findByText('this adventure is for people, but also animals');
  expect(container).toMatchSnapshot();
});
