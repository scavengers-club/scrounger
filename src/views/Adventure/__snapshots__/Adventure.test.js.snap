// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Adventure/> should render the list of adventures 1`] = `
<div>
  <div
    class="adventure-detail"
  >
    <h3>
      mount hood
    </h3>
    <img
      alt="mount hood"
      src="https://kfnfolzoauwbouvhqbrv.supabase.in/storage/v1/object/public/adv-images/trillium.jpg"
    />
    <h4>
      Description:
    </h4>
    <p>
      this adventure is for people, but also animals
    </p>
    <p>
      Hint 1: 
      hint 1
    </p>
    <p>
      Hint 2: 
      hint 2
    </p>
    <p>
      Hint 3: 
      hint 3
    </p>
    <h4>
      Solution: 
      Trillium Lake
    </h4>
  </div>
</div>
`;
